LuaRules/Gadgets/unit_targetting_metal_limit.lua

```diff
@@ -0,0 +1,202 @@
+-------------------------------------------------------------------------------------
+-- Value is the default state of the command
+local metalThreshholdsByUnitDefIDs = include("LuaRules/Configs/unit_targetting_metal_limit_defs.lua")
+
+local unitMetalMin = {}
+local unitDefCost = {}
+
+for i = 1, #UnitDefs do
+	local ud = UnitDefs[i]
+	unitDefCost[i] = ud.cost
+end
+
+function printThing(theKey, theTable, indent)
+	if indent == nil then
+		indent = ""
+	end
+	if (type(theTable) == "table") then
+		Spring.Echo(indent .. theKey .. ":")
+		for a, b in pairs(theTable) do
+			printThing(tostring(a), b, indent .. "  ")
+		end
+	else
+		Spring.Echo(indent .. theKey .. ": " .. tostring(theTable))
+	end
+end
+
+include("LuaRules/Configs/customcmds.h.lua")
+
+local preventChaffShootingCmdDesc = {
+	id      = CMD_MIN_METAL_TO_TARGET,
+	type    = CMDTYPE.ICON_MODE,
+	name    = "Min metal for kill.",
+	action  = 'preventchaffshootingmetallimit',
+	tooltip	= 'Enable to prevent units shooting at units which are very cheap.',
+	params 	= {0, 0, 100, 300, 1000}
+}
+
+function ChaffShootingPrevention_CheckMinMetal(unitID, targetID, damage)
+	Spring.Echo("Min metal check " .. tostring(Spring.GetGameFrame()))
+	if not (unitID and targetID and unitMetalMin[unitID]) then
+		return false
+	end
+
+	if spValidUnitID(unitID) and spValidUnitID(targetID) then
+		local gameFrame = spGetGameFrame()
+		local targetVisiblityState = spGetUnitLosState(targetID, Spring.GetUnitTeam(unitID), true)
+		local identified = (targetVisiblityState > 2)
+		Spring.Echo("testing")
+		if not identified then
+			Spring.Echo("Unidentified")
+			return unitMetalMin[unitID] > 0
+		end
+		local unitDefID = spGetUnitDefID(targetID)
+		Spring.Echo("testing2")
+		Spring.Echo("unitID" .. tostring(unitID))
+		Spring.Echo("unitDefIDAsKnown" .. tostring(unitDefIDAsKnown))
+		Spring.Echo("unitDefID" .. tostring(unitDefID))
+		Spring.Echo("unitMetalMin" .. tostring(unitMetalMin[unitID]))
+		Spring.Echo("unitDefCost" .. tostring(unitDefCost[unitDefIDAsKnown]))
+		if unitMetalMin[unitID] and unitMetalMin[unitID] > unitDefCost[unitDefID] then
+			return true
+		end
+	end
+	return false
+end
+
+--------------------------------------------------------------------------------
+-- Command Handling 
+local function PreventFiringAtChaffToggleCommand(unitID, unitDefID, cmdParams, cmdOptions)
+	if canHandleUnit[unitID] then
+		local state = cmdParams[1] or 1
+		local cmdDescID = spFindUnitCmdDesc(unitID, CMD_MIN_METAL_TO_TARGET)
+
+		local newParams = {}
+		if (cmdDescID) then
+			for k,v in pairs(metalThreshholdsByUnitDefIDs[unitDefID]) do newParams[k] = v end
+			newParams[1] = state
+			preventChaffShootingCmdDesc.params = newParams
+			spEditUnitCmdDesc(unitID, cmdDescID, {params = preventChaffShootingCmdDesc.params})
+		end
+		Spring.SendLuaRulesMsg(minMetalMsg .. " ".. tostring(unitID) .. " " .. tostring(metalThreshholdsByUnitDefIDs[unitDefID][state + 2]))
```

Comment:
Looks suspicious, why not just call the relevant thing directly?
```suggestion
		unitMetalMin[unitID] = metalThreshholdsByUnitDefIDs[unitDefID][state + 2]
```

Reply:
Can't remember what this was about...

Changed to your suggestion